pallets/rad-claims/src/lib.rs

```diff
@@ -0,0 +1,577 @@
+//! investments.
+//!
+//! - \[`Config`]
+//! - \[`Call`]
+//! - \[`Pallet`]
+//!
+//! ## Overview
+//! This pallet is used for processing reward payout claims from investors who
+//! invested in [Tinlake](https://tinlake.centrifuge.io) pools.
+//!
+//! ## Terminology
+//! RAD is the native token of Centrifuge chain.
+//! 
+//! ## Usage
+//!
+//! ## Interface
+//!
+//! ### Supported Origins
+//! Valid origin is an administrator or root.
+//!
+//! ### Types
+//! <code>\`AdminOrigin\`</code> Ensure that origin of a transaction is an administrator.
+//! <code>\`Currency\`</code> Expected currency of the reward claim.
+//! <code>\`Event\`</code> Overarching type for pallet events.
+//! <code>\`MinimalPayoutAmount\`</code> Minimal reward payout amount that can be claimed.
+//! <code>\`Longevity\`</code> Expected longevity of a validated unsigned extrinsic.
+//! <code>\`PalletId\`</code> Constant configuration parameter to store the module identifier for the pallet.
+//! <code>\`UnsignedPriority\`</code> A configuration for base priority of unsigned transactions.
+//! <code>\`WeightInfo\`</code> Pallet's transaction weights, calculated using runtime benchmarking.
+//!
+//! ### Events
+//!
+//! ### Errors
+//! <code>\`InsufficientBalance\`</code> Amount being claimed is less than the available amount in [AccountBalances].
+//! <code>\`InvalidProofs\`</code> The combination of account id, amount, and proofs vector in a claim was invalid.
+//! <code>\`MustBeAdmin\`</code> Protected operation, must be performed by admin.
+//! <code>\`UnderMinPayout\`</code> The payout amount attempting to be claimed is less than the minimum allowed by [MIN_PAYOUT].
+//!
+//! ### Dispatchable Functions
+//!
+//! Callable functions (or extrinsics), also considered as transactions, materialize the
+//! pallet contract. Here's the callable functions implemented in this module:
+//!
+//! [`claim`]
+//! [`set_upload_account`]
+//! [`store_root_hash`]
+//! 
+//! ### Public Functions
+//! [`sorted_hash_of`]
+//!
+//! ## Genesis Configuration
+//! The pallet is parameterized and configured via [parameter_types] macro, at the time the runtime is built
+//! by means of the [`construct_runtime`] macro.
+//!
+//! ## Dependencies
+//! This pallet is tightly coupled to:
+//! - Substrate FRAME's [balances pallet](https://github.com/paritytech/substrate/tree/master/frame/balances).
+//!
+//! ## References
+//! - [Substrate FRAME v2 attribute macros](https://crates.parity.io/frame_support/attr.pallet.html).
+//!
+//! ## Credits
+//! The Centrifugians Tribe <tribe@centrifuge.io>
+
+// Ensure we're `no_std` when compiling for WebAssembly.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+
+// ----------------------------------------------------------------------------
+// Imports and dependencies
+// ----------------------------------------------------------------------------
+
+// Mock runtime and unit test cases
+mod mock;
+mod tests;
```

Comment:
Do you mean to do this:
```rust
#![cfg(test)]
mod mock;

#![cfg(test)]
mod tests;
```
instead of placing `#![cfg(test)]` at the top of a module implementation?
What is your preference guys? Let's vote :) Personally I prefer to have that in the module implementation (very subjective, however).

Reply:
> Do you mean to do this

yes. Much more readable than the adding it to the top of the module implementation IMO

Reply:
Have to say, that I prefer this too. ^^

Reply:
Okay guys... you pocketed the bet :)